cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,172 @@
+            volume_size=image_config.root_volume.size,
+            volume_type="gp2",
+        )
+
+        cloudformation_dir = "../../../../cloudformation/imagebuilder/"
+
+        # Component
+        if dev_settings_config and dev_settings_config.update_and_reboot:
+            imagebuilder.CfnComponent(
+                self,
+                id="UpdateAndRebootComponent",
+                name="UpdateAndReboot",
+                version="0.0.1",
+                description="Update OS and Reboot",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "update_and_reboot.yaml")),
+            )
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster AMI",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "pcluster_install.yaml")),
+            )
+
+            components = [
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("UpdateAndReboot")
+                ),
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                ),
+            ]
+        else:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster custom AMI",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "pcluster_install.yaml")),
+            )
+
+            components = [
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                )
+            ]
+
+        # ImageRecipe
+        imagebuilder.CfnImageRecipe(
+            self,
+            id="PClusterImageRecipe",
+            name="PCluster-2-10-1" + str(hash(imagebuild)),
+            version="0.0.1",
+            parent_image=core.Fn.sub(
+                imagebuild.config.build_config.parent_image
+                or "arn:${AWS::Partition}:imagebuilder:${AWS::Region}:aws:image/amazon-linux-2-x86/x.x.x"
+            ),
+            components=components,
+            block_device_mappings=[
+                imagebuilder.CfnImageRecipe.InstanceBlockDeviceMappingProperty(
+                    device_name="/dev/xvda",
+                    ebs=ebs,
+                )
+            ],
+            tags={"ParallelClusterVersion": "2.10.1"},
```

Comment:
not sure if it was intended to tag the imageRecipe as well, but to tag the output image this should be set on the `imagebuilder.CfnImage`

Reply:
Deleted here, will add it if we need